### PR TITLE
142688 - Bess Configuration Service get enccellname from prefix patterns

### PR DESCRIPTION
**Pull request by MastekRohitChettri**
This PR consists changes of [PBI : 142688](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/142688)

### CODE REVIEW

**Review by harshal11869 — CHANGES_REQUESTED** [review]

**Review by shirinbt14926 — COMMENTED** [review]
Minor comments please

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs [hunk 1]
```diff
@@ -603,7 +603,15 @@ public enum EventIds
         /// <summary>
         /// 809119 - Bess Builder Service Completed
         /// </summary>
-        BessBuilderServiceCompleted = 809119
+        BessBuilderServiceCompleted = 809119,
+        /// <summary>
+        /// 809120 - Listed cells or pattern not found in catalog
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Logging/EventIds.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs [hunk 2]
```diff
@@ -245,5 +262,59 @@ private ScheduleDetailEntity GetScheduleDetail(DateTime nextOccurrence, BessConf
...
+        /// <summary>
+        /// Fetched the list of all the ENC cell names from the provided prefix patterns
+        /// AIO cell is excluded
+        /// </summary>
+        /// <param name="encCellNames"></param>
+        /// <param name="salesCatalogueProducts"></param>
+        /// <returns></returns>
+        private IEnumerable<(string, int?)> GetEncCells(IEnumerable<string> encCellNames, IEnumerable<SalesCatalogueDataProductResponse> salesCatalogueProducts)
+        {
+            #region filter provided prefix patterns
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/UKHO.PeriodicOutputService.Common.csproj [hunk 3]
```diff
@@ -23,8 +23,7 @@
     <PackageReference Include="Azure.Storage.Blobs" Version="12.19.1" />
     <PackageReference Include="DiscUtils.Iso9660" Version="0.16.13" />
     <PackageReference Include="Microsoft.Extensions.Caching.Abstractions" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Http" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Http.Polly" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Http.Polly" Version="8.0.3" />
```
[the remark this answers is not in the document]
Comment (MastekRohitChettri) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/UKHO.PeriodicOutputService.Common.csproj:26]: Got issues with packages when took develop branch so reinstall latest package also there is unnecessary package was used which we removed (<PackageReference Include="Microsoft.Extensions.Http" Version="7.0.0" />)

**Review by harshal11869 — APPROVED** [review]
Looks Good.

**Review by Nadeem14456 — APPROVED** [review]
LGTM.

**Review by DanOwenUKHO — APPROVED** [review]

**Review by shirinbt14926 — APPROVED** [review]
Looks good

**Pull request merged**